Author: Daniel Baumann <mail@daniel-baumann.ch>
Description:
 Fixing symbol conflicts with other MD5 implementations (Closes: #522512).

--- a/src/md5.h
+++ b/src/md5.h
@@ -21,6 +21,15 @@
 #ifndef LIBDVDREAD_MD5_H
 #define LIBDVDREAD_MD5_H
 
+/* Change function names to prevent symbol conflicts */
+#define md5_init_ctx _DVD_md5_init_ctx
+#define md5_process_block _DVD_md5_process_block
+#define md5_process_bytes _DVD_md5_process_bytes
+#define md5_finish_ctx _DVD_md5_finish_ctx
+#define md5_read_ctx _DVD_md5_read_ctx
+#define md5_stream _DVD_md5_stream
+#define md5_buffer _DVD_md5_buffer
+
 #include <stdio.h>
 
 #if defined HAVE_LIMITS_H || defined _LIBC
