diff -Naur a/xine_frontend_cec.c b/xine_frontend_cec.c
--- a/xine_frontend_cec.c	2014-12-20 06:04:26.000000000 +0200
+++ b/xine_frontend_cec.c	2014-12-25 16:34:07.391229903 +0200
@@ -272,7 +272,7 @@
   p->bMonitorOnly = 0;
   p->cecVersion = CEC_DEFAULT_SETTING_CEC_VERSION;
   p->adapterType = ADAPTERTYPE_UNKNOWN;
-  p->iDoubleTapTimeoutMs = CEC_DOUBLE_TAP_TIMEOUT_MS;
+  p->iDoubleTapTimeout50Ms = CEC_DOUBLE_TAP_TIMEOUT_50_MS;
   p->comboKey = CEC_USER_CONTROL_CODE_STOP;
   p->iComboKeyTimeoutMs = CEC_DEFAULT_COMBO_TIMEOUT_MS;
 
