diff -Naur a/Makefile b/Makefile
--- a/Makefile	2015-06-25 19:55:21.000000000 +0300
+++ b/Makefile	2015-06-25 20:07:13.714992078 +0300
@@ -32,7 +32,7 @@
 ifeq ($(CONFIG_W_UNUSED_RESULT),yes)
 CFLAGS  += -Wunused-result
 endif
-CFLAGS  += -Wall -Werror -Wwrite-strings -Wno-deprecated-declarations
+CFLAGS  += -Wall -Wwrite-strings -Wno-deprecated-declarations
 CFLAGS  += -Wmissing-prototypes
 CFLAGS  += -fms-extensions -funsigned-char -fno-strict-aliasing
 CFLAGS  += -D_FILE_OFFSET_BITS=64
