diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-03-14 04:42:30.000000000 +0200
+++ b/Makefile	2016-03-14 12:06:17.261924193 +0200
@@ -32,9 +32,9 @@
 ifeq ($(CONFIG_W_UNUSED_RESULT),yes)
 CFLAGS  += -Wunused-result
 endif
-ifneq ($(CFLAGS_NO_WERROR),yes)
-CFLAGS  += -Werror
-endif
+#ifneq ($(CFLAGS_NO_WERROR),yes)
+#CFLAGS  += -Werror
+#endif
 CFLAGS  += -Wall -Wwrite-strings -Wno-deprecated-declarations
 CFLAGS  += -Wmissing-prototypes
 CFLAGS  += -fms-extensions -funsigned-char -fno-strict-aliasing
diff -Naur a/Makefile.ffmpeg b/Makefile.ffmpeg
--- a/Makefile.ffmpeg	2016-03-14 04:42:30.000000000 +0200
+++ b/Makefile.ffmpeg	2016-03-14 13:55:43.597640940 +0200
@@ -290,13 +290,13 @@
 	cd $(LIB_ROOT)/$(LIBX264) && $(CONFIGURE) \
 		--prefix=/ffmpeg \
 		--enable-static \
-		--disable-shared \
 		--disable-avs \
 		--disable-swscale \
 		--disable-lavf \
 		--disable-ffms \
 		--disable-gpac \
 		--disable-lsmash \
+		--disable-asm
 		$(LIBX264_HOST)
 	DESTDIR=$(LIB_ROOT)/build \
 		$(MAKE) -C $(LIB_ROOT)/$(LIBX264) install
@@ -362,7 +362,11 @@
 	cd $(LIB_ROOT)/$(LIBVPX) && $(CONFIGURE) \
 		--prefix=/ffmpeg \
 		--enable-static \
-		--disable-shared
+		--disable-shared \
+		--as=yasm \
+		--disable-examples \
+		--disable-docs \
+		--disable-unit-tests
 	DIST_DIR=$(LIB_ROOT)/build/ffmpeg \
 		$(MAKE) -C $(LIB_ROOT)/$(LIBVPX) install
 	@touch $@
