diff -Naur a/xbmc/pvr/windows/GUIWindowPVRSearch.cpp b/xbmc/pvr/windows/GUIWindowPVRSearch.cpp
--- a/xbmc/pvr/windows/GUIWindowPVRSearch.cpp	2016-04-26 10:59:06.000000000 +0300
+++ b/xbmc/pvr/windows/GUIWindowPVRSearch.cpp	2016-05-09 01:19:28.446054335 +0300
@@ -159,6 +159,7 @@
 
     bAddSpecialSearchItem = true;
 
+    items.Clear();
     CGUIDialogProgress* dlgProgress = (CGUIDialogProgress*)g_windowManager.GetWindow(WINDOW_DIALOG_PROGRESS);
     if (dlgProgress)
     {
