diff -Naur a/source/mupen64plus-audio-sdl/projects/unix/Makefile b/source/mupen64plus-audio-sdl/projects/unix/Makefile
--- a/source/mupen64plus-audio-sdl/projects/unix/Makefile	2013-07-04 03:28:02.000000000 +0300
+++ b/source/mupen64plus-audio-sdl/projects/unix/Makefile	2015-02-01 14:19:43.192082113 +0200
@@ -264,8 +264,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-RM       ?= rm -f
+#CC        = $(CROSS_COMPILE)gcc
+#RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
 COMPILE.c = $(Q_CC)$(CC) $(CFLAGS) $(CPPFLAGS) $(TARGET_ARCH) -c
diff -Naur a/source/mupen64plus-audio-sdl/src/main.c b/source/mupen64plus-audio-sdl/src/main.c
--- a/source/mupen64plus-audio-sdl/src/main.c	2013-07-04 03:28:02.000000000 +0300
+++ b/source/mupen64plus-audio-sdl/src/main.c	2015-02-01 17:51:28.339791607 +0200
@@ -26,8 +26,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <SDL.h>
-#include <SDL_audio.h>
+#include <SDL/SDL.h>
+#include <SDL/SDL_audio.h>
 
 #ifdef USE_SRC
 #include <samplerate.h>
diff -Naur a/source/mupen64plus-core/projects/unix/Makefile b/source/mupen64plus-core/projects/unix/Makefile
--- a/source/mupen64plus-core/projects/unix/Makefile	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/projects/unix/Makefile	2015-02-01 14:19:23.632082560 +0200
@@ -284,8 +284,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
+#CC        = $(CROSS_COMPILE)gcc
+#CXX       = $(CROSS_COMPILE)g++
 RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
diff -Naur a/source/mupen64plus-core/src/api/frontend.c b/source/mupen64plus-core/src/api/frontend.c
--- a/source/mupen64plus-core/src/api/frontend.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/api/frontend.c	2015-02-01 17:54:08.051787955 +0200
@@ -27,7 +27,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #define M64P_CORE_PROTOTYPES 1
 #include "m64p_types.h"
diff -Naur a/source/mupen64plus-core/src/api/vidext.c b/source/mupen64plus-core/src/api/vidext.c
--- a/source/mupen64plus-core/src/api/vidext.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/api/vidext.c	2015-02-01 17:55:38.523785886 +0200
@@ -25,7 +25,7 @@
 
 #include <stdlib.h>
 #include <string.h>
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #define M64P_CORE_PROTOTYPES 1
 #include "m64p_types.h"
diff -Naur a/source/mupen64plus-core/src/api/vidext_sdl2_compat.h b/source/mupen64plus-core/src/api/vidext_sdl2_compat.h
--- a/source/mupen64plus-core/src/api/vidext_sdl2_compat.h	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/api/vidext_sdl2_compat.h	2015-02-01 18:02:06.207777022 +0200
@@ -19,7 +19,7 @@
   3. This notice may not be removed or altered from any source distribution.
 */
 
-#include <SDL_surface.h>
+#include <SDL/SDL_surface.h>
 
 typedef struct
 {
diff -Naur a/source/mupen64plus-core/src/debugger/dbg_breakpoints.c b/source/mupen64plus-core/src/debugger/dbg_breakpoints.c
--- a/source/mupen64plus-core/src/debugger/dbg_breakpoints.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/debugger/dbg_breakpoints.c	2015-02-01 17:56:22.531784880 +0200
@@ -19,8 +19,8 @@
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
-#include <SDL.h>
-#include <SDL_thread.h>
+#include <SDL/SDL.h>
+#include <SDL/SDL_thread.h>
 
 #include "dbg_types.h"
 #include "debugger.h"
diff -Naur a/source/mupen64plus-core/src/debugger/debugger.c b/source/mupen64plus-core/src/debugger/debugger.c
--- a/source/mupen64plus-core/src/debugger/debugger.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/debugger/debugger.c	2015-02-01 17:57:47.607782935 +0200
@@ -20,7 +20,7 @@
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #include "api/debugger.h"
 
diff -Naur a/source/mupen64plus-core/src/main/cheat.c b/source/mupen64plus-core/src/main/cheat.c
--- a/source/mupen64plus-core/src/main/cheat.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/cheat.c	2015-02-01 18:04:18.435773998 +0200
@@ -22,8 +22,8 @@
 
 // gameshark and xploder64 reference: http://doc.kodewerx.net/hacking_n64.html 
 
-#include <SDL.h>
-#include <SDL_thread.h>
+#include <SDL/SDL.h>
+#include <SDL/SDL_thread.h>
 
 #include "api/m64p_types.h"
 #include "api/callbacks.h"
diff -Naur a/source/mupen64plus-core/src/main/eventloop.c b/source/mupen64plus-core/src/main/eventloop.c
--- a/source/mupen64plus-core/src/main/eventloop.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/eventloop.c	2015-02-01 18:04:54.139773182 +0200
@@ -21,7 +21,7 @@
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
 #include <stdlib.h>
-#include <SDL.h>
+#include <SDL/SDL.h>
 #if ! SDL_VERSION_ATLEAST(1,3,0)
 
 #define SDL_SCANCODE_ESCAPE SDLK_ESCAPE
diff -Naur a/source/mupen64plus-core/src/main/main.c b/source/mupen64plus-core/src/main/main.c
--- a/source/mupen64plus-core/src/main/main.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/main.c	2015-02-01 18:05:30.827772343 +0200
@@ -31,7 +31,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #define M64P_CORE_PROTOTYPES 1
 #include "api/m64p_types.h"
diff -Naur a/source/mupen64plus-core/src/main/savestates.c b/source/mupen64plus-core/src/main/savestates.c
--- a/source/mupen64plus-core/src/main/savestates.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/savestates.c	2015-02-01 18:06:11.187771420 +0200
@@ -24,7 +24,7 @@
 
 #include <stdlib.h>
 #include <string.h>
-#include <SDL_thread.h>
+#include <SDL/SDL_thread.h>
 
 #define M64P_CORE_PROTOTYPES 1
 #include "api/m64p_types.h"
diff -Naur a/source/mupen64plus-core/src/main/workqueue.c b/source/mupen64plus-core/src/main/workqueue.c
--- a/source/mupen64plus-core/src/main/workqueue.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/workqueue.c	2015-02-01 18:06:40.019770761 +0200
@@ -22,8 +22,8 @@
 #include "workqueue.h"
 #include "api/callbacks.h"
 
-#include <SDL.h>
-#include <SDL_thread.h>
+#include <SDL/SDL.h>
+#include <SDL/SDL_thread.h>
 
 #define WORKQUEUE_THREADS 1
 
diff -Naur a/source/mupen64plus-core/src/main/zip/ioapi.h b/source/mupen64plus-core/src/main/zip/ioapi.h
--- a/source/mupen64plus-core/src/main/zip/ioapi.h	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/zip/ioapi.h	2015-02-01 19:41:16.383640969 +0200
@@ -43,7 +43,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include "zlib.h"
+#include <zlib.h>
 
 #ifndef OF
 #define OF _Z_OF
diff -Naur a/source/mupen64plus-core/src/main/zip/unzip.c b/source/mupen64plus-core/src/main/zip/unzip.c
--- a/source/mupen64plus-core/src/main/zip/unzip.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/zip/unzip.c	2015-02-01 19:41:39.463640442 +0200
@@ -72,7 +72,7 @@
         #define NOUNCRYPT
 #endif
 
-#include "zlib.h"
+#include <zlib.h>
 #include "unzip.h"
 
 #ifdef STDC
diff -Naur a/source/mupen64plus-core/src/main/zip/unzip.h b/source/mupen64plus-core/src/main/zip/unzip.h
--- a/source/mupen64plus-core/src/main/zip/unzip.h	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/zip/unzip.h	2015-02-01 19:37:50.107645686 +0200
@@ -48,7 +48,7 @@
 #endif
 
 #ifndef _ZLIB_H
-#include "zlib.h"
+#include <zlib.h>
 #endif
 
 #ifndef  _ZLIBIOAPI_H
diff -Naur a/source/mupen64plus-core/src/main/zip/zip.c b/source/mupen64plus-core/src/main/zip/zip.c
--- a/source/mupen64plus-core/src/main/zip/zip.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/zip/zip.c	2015-02-01 19:42:15.243639624 +0200
@@ -26,7 +26,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <time.h>
-#include "zlib.h"
+#include <zlib.h>
 #include "zip.h"
 
 #ifdef STDC
diff -Naur a/source/mupen64plus-core/src/main/zip/zip.h b/source/mupen64plus-core/src/main/zip/zip.h
--- a/source/mupen64plus-core/src/main/zip/zip.h	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/main/zip/zip.h	2015-02-01 19:42:33.135639215 +0200
@@ -47,7 +47,7 @@
 //#define HAVE_BZIP2
 
 #ifndef _ZLIB_H
-#include "zlib.h"
+#include <zlib.h>
 #endif
 
 #ifndef _ZLIBIOAPI_H
diff -Naur a/source/mupen64plus-core/src/osd/OGLFT.h b/source/mupen64plus-core/src/osd/OGLFT.h
--- a/source/mupen64plus-core/src/osd/OGLFT.h	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/osd/OGLFT.h	2015-02-01 18:08:47.627767843 +0200
@@ -30,7 +30,7 @@
 #include <wchar.h>
 
 #define GL_GLEXT_PROTOTYPES
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 #if defined(__MACOSX__)
 #include <OpenGL/glu.h>
 #elif defined(__MACOS__)
diff -Naur a/source/mupen64plus-core/src/osd/osd.cpp b/source/mupen64plus-core/src/osd/osd.cpp
--- a/source/mupen64plus-core/src/osd/osd.cpp	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/osd/osd.cpp	2015-02-01 18:09:21.859767060 +0200
@@ -20,8 +20,8 @@
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
 // On-screen Display
-#include <SDL_opengl.h>
-#include <SDL_thread.h>
+#include <SDL/SDL_opengl.h>
+#include <SDL/SDL_thread.h>
 
 #include "OGLFT.h"
 #include "osd.h"
diff -Naur a/source/mupen64plus-core/src/osd/screenshot.cpp b/source/mupen64plus-core/src/osd/screenshot.cpp
--- a/source/mupen64plus-core/src/osd/screenshot.cpp	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/osd/screenshot.cpp	2015-02-01 18:09:56.419766270 +0200
@@ -24,8 +24,8 @@
 #include <stdio.h>
 #include <ctype.h>
 
-#include <SDL_opengl.h>
-#include <SDL.h>
+#include <SDL/SDL_opengl.h>
+#include <SDL/SDL.h>
 #include <png.h>
 
 #include "osd.h"
diff -Naur a/source/mupen64plus-core/src/r4300/interupt.c b/source/mupen64plus-core/src/r4300/interupt.c
--- a/source/mupen64plus-core/src/r4300/interupt.c	2013-07-04 03:27:56.000000000 +0300
+++ b/source/mupen64plus-core/src/r4300/interupt.c	2015-02-01 18:11:04.363764717 +0200
@@ -21,7 +21,7 @@
 
 #include <stdlib.h>
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #define M64P_CORE_PROTOTYPES 1
 #include "api/m64p_types.h"
diff -Naur a/source/mupen64plus-input-sdl/projects/unix/Makefile b/source/mupen64plus-input-sdl/projects/unix/Makefile
--- a/source/mupen64plus-input-sdl/projects/unix/Makefile	2013-07-04 03:28:05.000000000 +0300
+++ b/source/mupen64plus-input-sdl/projects/unix/Makefile	2015-02-01 14:20:02.712081666 +0200
@@ -214,8 +214,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
+#CC        = $(CROSS_COMPILE)gcc
+#CXX       = $(CROSS_COMPILE)g++
 RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
diff -Naur a/source/mupen64plus-input-sdl/src/config.c b/source/mupen64plus-input-sdl/src/config.c
--- a/source/mupen64plus-input-sdl/src/config.c	2013-07-04 03:28:05.000000000 +0300
+++ b/source/mupen64plus-input-sdl/src/config.c	2015-02-01 18:18:44.683754191 +0200
@@ -19,7 +19,7 @@
  *   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.          *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #define M64P_PLUGIN_PROTOTYPES 1
 #include "m64p_types.h"
diff -Naur a/source/mupen64plus-input-sdl/src/plugin.c b/source/mupen64plus-input-sdl/src/plugin.c
--- a/source/mupen64plus-input-sdl/src/plugin.c	2013-07-04 03:28:05.000000000 +0300
+++ b/source/mupen64plus-input-sdl/src/plugin.c	2015-02-01 18:19:03.691753757 +0200
@@ -25,7 +25,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #define M64P_PLUGIN_PROTOTYPES 1
 #include "m64p_types.h"
diff -Naur a/source/mupen64plus-input-sdl/src/plugin.h b/source/mupen64plus-input-sdl/src/plugin.h
--- a/source/mupen64plus-input-sdl/src/plugin.h	2013-07-04 03:28:05.000000000 +0300
+++ b/source/mupen64plus-input-sdl/src/plugin.h	2015-02-01 18:19:17.315753445 +0200
@@ -24,7 +24,7 @@
 #ifndef __PLUGIN_H__
 #define __PLUGIN_H__
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 #if ! SDL_VERSION_ATLEAST(1,3,0)
 
 #define SDL_GetKeyboardState SDL_GetKeyState
diff -Naur a/source/mupen64plus-rsp-hle/projects/unix/Makefile b/source/mupen64plus-rsp-hle/projects/unix/Makefile
--- a/source/mupen64plus-rsp-hle/projects/unix/Makefile	2013-07-04 03:28:07.000000000 +0300
+++ b/source/mupen64plus-rsp-hle/projects/unix/Makefile	2015-02-01 14:20:23.344081195 +0200
@@ -203,8 +203,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
+#CC        = $(CROSS_COMPILE)gcc
+#CXX       = $(CROSS_COMPILE)g++
 RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
diff -Naur a/source/mupen64plus-ui-console/projects/unix/Makefile b/source/mupen64plus-ui-console/projects/unix/Makefile
--- a/source/mupen64plus-ui-console/projects/unix/Makefile	2013-07-04 03:28:00.000000000 +0300
+++ b/source/mupen64plus-ui-console/projects/unix/Makefile	2015-02-01 14:20:40.900080793 +0200
@@ -197,8 +197,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
+#CC        = $(CROSS_COMPILE)gcc
+#CXX       = $(CROSS_COMPILE)g++
 RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
diff -Naur a/source/mupen64plus-ui-console/src/main.c b/source/mupen64plus-ui-console/src/main.c
--- a/source/mupen64plus-ui-console/src/main.c	2013-07-04 03:28:00.000000000 +0300
+++ b/source/mupen64plus-ui-console/src/main.c	2015-02-01 18:20:57.487751155 +0200
@@ -30,7 +30,7 @@
 #include <stdlib.h>
 #include <stdarg.h>
 
-#include <SDL_main.h>
+#include <SDL/SDL_main.h>
 
 #include "cheat.h"
 #include "main.h"
diff -Naur a/source/mupen64plus-video-glide64mk2/projects/unix/Makefile b/source/mupen64plus-video-glide64mk2/projects/unix/Makefile
--- a/source/mupen64plus-video-glide64mk2/projects/unix/Makefile	2013-07-04 03:28:17.000000000 +0300
+++ b/source/mupen64plus-video-glide64mk2/projects/unix/Makefile	2015-02-01 14:20:56.156080444 +0200
@@ -292,8 +292,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
+#CC        = $(CROSS_COMPILE)gcc
+#CXX       = $(CROSS_COMPILE)g++
 RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
diff -Naur a/source/mupen64plus-video-glide64mk2/src/Glide64/Gfx_1.3.h b/source/mupen64plus-video-glide64mk2/src/Glide64/Gfx_1.3.h
--- a/source/mupen64plus-video-glide64mk2/src/Glide64/Gfx_1.3.h	2013-07-04 03:28:17.000000000 +0300
+++ b/source/mupen64plus-video-glide64mk2/src/Glide64/Gfx_1.3.h	2015-02-01 18:22:40.563748798 +0200
@@ -66,7 +66,7 @@
 #include "m64p.h"
 
 #include <stdio.h>
-#include <SDL_mutex.h>
+#include <SDL/SDL_mutex.h>
 #include <fstream>
 #include <stdlib.h>
 #include <stddef.h>		// offsetof
diff -Naur a/source/mupen64plus-video-glide64mk2/src/Glide64/TexCache.cpp b/source/mupen64plus-video-glide64mk2/src/Glide64/TexCache.cpp
--- a/source/mupen64plus-video-glide64mk2/src/Glide64/TexCache.cpp	2013-07-04 03:28:17.000000000 +0300
+++ b/source/mupen64plus-video-glide64mk2/src/Glide64/TexCache.cpp	2015-02-01 18:24:01.395746950 +0200
@@ -37,7 +37,7 @@
 //
 //****************************************************************
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 #include "Gfx_1.3.h"
 #include "TexCache.h"
 #include "Combine.h"
diff -Naur a/source/mupen64plus-video-glide64mk2/src/GlideHQ/TxHiResCache.cpp b/source/mupen64plus-video-glide64mk2/src/GlideHQ/TxHiResCache.cpp
--- a/source/mupen64plus-video-glide64mk2/src/GlideHQ/TxHiResCache.cpp	2013-07-04 03:28:17.000000000 +0300
+++ b/source/mupen64plus-video-glide64mk2/src/GlideHQ/TxHiResCache.cpp	2015-02-01 18:24:45.503745941 +0200
@@ -54,7 +54,7 @@
 
 #include <zlib.h>
 #include <string>
-#include <SDL.h>
+#include <SDL/SDL.h>
 #include "TxHiResCache.h"
 #include "TxDbg.h"
 #include "../Glide64/Gfx_1.3.h"
diff -Naur a/source/mupen64plus-video-glide64mk2/src/Glitch64/glitchmain.cpp b/source/mupen64plus-video-glide64mk2/src/Glitch64/glitchmain.cpp
--- a/source/mupen64plus-video-glide64mk2/src/Glitch64/glitchmain.cpp	2013-07-04 03:28:17.000000000 +0300
+++ b/source/mupen64plus-video-glide64mk2/src/Glitch64/glitchmain.cpp	2015-02-01 18:25:30.079744922 +0200
@@ -27,7 +27,7 @@
 #include <stdint.h>
 #include <stdarg.h>
 #include <string.h>
-#include <SDL.h>
+#include <SDL/SDL.h>
 #endif // _WIN32
 #include <stdlib.h>
 #include <stdio.h>
diff -Naur a/source/mupen64plus-video-glide64mk2/src/Glitch64/main.h b/source/mupen64plus-video-glide64mk2/src/Glitch64/main.h
--- a/source/mupen64plus-video-glide64mk2/src/Glitch64/main.h	2013-07-04 03:28:17.000000000 +0300
+++ b/source/mupen64plus-video-glide64mk2/src/Glitch64/main.h	2015-02-01 18:26:22.415743725 +0200
@@ -67,7 +67,7 @@
 #ifdef _WIN32
 #include <windows.h>
 extern "C" {
-    #include <SDL_opengl.h>
+    #include <SDL/SDL_opengl.h>
     extern PFNGLACTIVETEXTUREARBPROC glActiveTextureARB;
     extern PFNGLATTACHOBJECTARBPROC glAttachObjectARB;
     extern PFNGLBINDFRAMEBUFFEREXTPROC glBindFramebufferEXT;
@@ -102,7 +102,7 @@
 #include <stdio.h>
 //#define printf(...)
 #define GL_GLEXT_PROTOTYPES
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 #endif // _WIN32
 #include "glide.h"
 
diff -Naur a/source/mupen64plus-video-rice/projects/unix/Makefile b/source/mupen64plus-video-rice/projects/unix/Makefile
--- a/source/mupen64plus-video-rice/projects/unix/Makefile	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/projects/unix/Makefile	2015-02-01 14:21:16.180079986 +0200
@@ -261,8 +261,8 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
+#CC        = $(CROSS_COMPILE)gcc
+#CXX       = $(CROSS_COMPILE)g++
 RM       ?= rm -f
 INSTALL  ?= install
 MKDIR ?= mkdir -p
diff -Naur a/source/mupen64plus-video-rice/src/CNvTNTCombiner.cpp b/source/mupen64plus-video-rice/src/CNvTNTCombiner.cpp
--- a/source/mupen64plus-video-rice/src/CNvTNTCombiner.cpp	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/CNvTNTCombiner.cpp	2015-02-01 18:28:01.715741455 +0200
@@ -16,7 +16,7 @@
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */
 
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 
 #include "CNvTNTCombiner.h"
 
diff -Naur a/source/mupen64plus-video-rice/src/CritSect.h b/source/mupen64plus-video-rice/src/CritSect.h
--- a/source/mupen64plus-video-rice/src/CritSect.h	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/CritSect.h	2015-02-01 18:28:24.723740928 +0200
@@ -20,7 +20,7 @@
 #if !defined(CRITSECT_H)
 #define CRITSECT_H
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 class CCritSect
 {
diff -Naur a/source/mupen64plus-video-rice/src/OGLCombinerNV.h b/source/mupen64plus-video-rice/src/OGLCombinerNV.h
--- a/source/mupen64plus-video-rice/src/OGLCombinerNV.h	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/OGLCombinerNV.h	2015-02-01 18:28:52.875740285 +0200
@@ -21,7 +21,7 @@
 
 #include <vector>
 
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 
 #include "OGLExtCombiner.h"
 #include "OGLDecodedMux.h"
diff -Naur a/source/mupen64plus-video-rice/src/OGLCombinerTNT2.h b/source/mupen64plus-video-rice/src/OGLCombinerTNT2.h
--- a/source/mupen64plus-video-rice/src/OGLCombinerTNT2.h	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/OGLCombinerTNT2.h	2015-02-01 18:29:21.279739635 +0200
@@ -19,7 +19,7 @@
 #ifndef _OGL_COMBINER_TNT2_H_
 #define _OGL_COMBINER_TNT2_H_
 
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 
 #define GL_SOURCE3_RGB_EXT                0x8583
 #define GL_SOURCE3_ALPHA_EXT              0x858B
diff -Naur a/source/mupen64plus-video-rice/src/OGLExtensions.cpp b/source/mupen64plus-video-rice/src/OGLExtensions.cpp
--- a/source/mupen64plus-video-rice/src/OGLExtensions.cpp	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/OGLExtensions.cpp	2015-02-01 18:30:38.295737874 +0200
@@ -19,7 +19,7 @@
 /* This source file contains code for assigning function pointers to some OpenGL functions */
 /* This is only necessary because Windows does not contain development support for OpenGL versions beyond 1.1 */
 
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 #include "OGLExtensions.h"
 #include "Video.h"
 
diff -Naur a/source/mupen64plus-video-rice/src/OGLExtensions.h b/source/mupen64plus-video-rice/src/OGLExtensions.h
--- a/source/mupen64plus-video-rice/src/OGLExtensions.h	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/OGLExtensions.h	2015-02-01 18:30:02.187738700 +0200
@@ -22,7 +22,7 @@
 #if !defined(OGL_EXTENSIONS_H)
 #define OGL_EXTENSIONS_H
 
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 
 /* Just call this one function to load up the function pointers. */
 void OGLExtensions_Init(void);
diff -Naur a/source/mupen64plus-video-rice/src/OGLFragmentShaders.h b/source/mupen64plus-video-rice/src/OGLFragmentShaders.h
--- a/source/mupen64plus-video-rice/src/OGLFragmentShaders.h	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/OGLFragmentShaders.h	2015-02-01 18:31:03.263737303 +0200
@@ -21,7 +21,7 @@
 
 #include <vector>
 
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 
 #include "OGLCombiner.h"
 #include "OGLExtCombiner.h"
diff -Naur a/source/mupen64plus-video-rice/src/osal_opengl.h b/source/mupen64plus-video-rice/src/osal_opengl.h
--- a/source/mupen64plus-video-rice/src/osal_opengl.h	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/osal_opengl.h	2015-02-01 18:32:21.119735523 +0200
@@ -22,10 +22,10 @@
 #if !defined(OSAL_OPENGL_H)
 #define OSAL_OPENGL_H
 
-#include <SDL_config.h>
+#include <SDL/SDL_config.h>
 
 #if SDL_VIDEO_OPENGL
-#include <SDL_opengl.h>
+#include <SDL/SDL_opengl.h>
 #define GLSL_VERSION "120"
 
 // Extension names
@@ -35,7 +35,7 @@
 
 
 #elif SDL_VIDEO_OPENGL_ES2
-#include <SDL_opengles2.h>
+#include <SDL/SDL_opengles2.h>
 #define GLSL_VERSION "100"
 
 // Extension names
diff -Naur a/source/mupen64plus-video-rice/src/RSP_Parser.cpp b/source/mupen64plus-video-rice/src/RSP_Parser.cpp
--- a/source/mupen64plus-video-rice/src/RSP_Parser.cpp	2013-07-04 03:28:10.000000000 +0300
+++ b/source/mupen64plus-video-rice/src/RSP_Parser.cpp	2015-02-01 18:33:32.971733880 +0200
@@ -18,7 +18,7 @@
 */
 
 #include <algorithm>
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #include "ConvertImage.h"
 #include "GraphicsContext.h"
