diff -ru a/linux-tbs-drivers/linux/drivers/media/dvb/frontends/stv090x.c b/linux-tbs-drivers/linux/drivers/media/dvb/frontends/stv090x.c
--- a/linux-tbs-drivers/linux/drivers/media/dvb/frontends/stv090x.c	2011-10-09 23:15:32.000000000 +0300
+++ b/linux-tbs-drivers/linux/drivers/media/dvb/frontends/stv090x.c	2013-03-28 20:21:49.953700510 +0200
@@ -3633,8 +3684,7 @@
 	u32 reg_0, reg_1, reg, i;
 	s32 val_0, val_1, val = 0;
 	u8 lock_f;
-	s32 div;
-	u32 last;
+	s32 snr;
 
 	switch (state->delsys) {
 	case STV090x_DVBS2:
@@ -3651,10 +3701,11 @@
 				msleep(1);
 			}
 			val /= 16;
-			last = ARRAY_SIZE(stv090x_s2cn_tab) - 1;
-			div = stv090x_s2cn_tab[0].read -
-			      stv090x_s2cn_tab[last].read;
-			*cnr = 0xFFFF - ((val * 0xFFFF) / div);
+			snr = stv090x_table_lookup(stv090x_s2cn_tab,
+				ARRAY_SIZE(stv090x_s2cn_tab) - 1, val);
+			if (snr < 0) snr = 0;
+			if (snr > 200) snr = 200;
+			*cnr = snr * 0xFFFF / 200;
 		}
 		break;
 
@@ -3673,10 +3724,11 @@
 				msleep(1);
 			}
 			val /= 16;
-			last = ARRAY_SIZE(stv090x_s1cn_tab) - 1;
-			div = stv090x_s1cn_tab[0].read -
-			      stv090x_s1cn_tab[last].read;
-			*cnr = 0xFFFF - ((val * 0xFFFF) / div);
+			snr = stv090x_table_lookup(stv090x_s1cn_tab,
+				ARRAY_SIZE(stv090x_s1cn_tab) - 1, val);
+			if (snr < 0) snr = 0;
+			if (snr > 200) snr = 200;
+			*cnr = snr * 0xFFFF / 200;
 		}
 		break;
 	default:
