diff -ru a/linux-tbs-drivers/linux/include/linux/dvb/frontend.h b/linux-tbs-drivers/linux/include/linux/dvb/frontend.h
--- a/linux-tbs-drivers/linux/include/linux/dvb/frontend.h	2011-05-21 02:59:30.000000000 +0300
+++ b/linux-tbs-drivers/linux/include/linux/dvb/frontend.h	2013-01-25 19:44:59.507326001 +0200
@@ -62,6 +62,7 @@
 	FE_CAN_8VSB			= 0x200000,
 	FE_CAN_16VSB			= 0x400000,
 	FE_HAS_EXTENDED_CAPS		= 0x800000,   /* We need more bitspace for newer APIs, indicate this. */
+	FE_CAN_MULTISTREAM		= 0x4000000,  /* frontend supports DVB-S2 multistream filtering */
 	FE_CAN_TURBO_FEC		= 0x8000000,  /* frontend supports "turbo fec modulation" */
 	FE_CAN_2G_MODULATION		= 0x10000000, /* frontend supports "2nd generation modulation" (DVB-S2) */
 	FE_NEEDS_BENDING		= 0x20000000, /* not supported anymore, don't use (frontend requires frequency bending) */
@@ -315,9 +316,12 @@
 #define DTV_ISDBS_TS_ID		42
 
 #define DTV_DVBT2_PLP_ID	43
+#define DTV_DVBS2_MIS_ID	43
 
 #define DTV_MAX_COMMAND				DTV_DVBT2_PLP_ID
 
+#define NO_STREAM_ID_FILTER	(~0U)
+
 typedef enum fe_pilot {
 	PILOT_ON,
 	PILOT_OFF,
diff -ru a/linux-tbs-drivers/linux/drivers/media/dvb/dvb-core/dvb_frontend.c b/linux-tbs-drivers/linux/drivers/media/dvb/dvb-core/dvb_frontend.c
--- a/linux-tbs-drivers/linux/drivers/media/dvb/dvb-core/dvb_frontend.c	2011-10-09 22:20:56.000000000 +0300
+++ b/linux-tbs-drivers/linux/drivers/media/dvb/dvb-core/dvb_frontend.c	2013-01-17 20:28:24.205948900 +0200
@@ -891,6 +891,7 @@
 		c->layer[i].interleaving = -1;
 		c->layer[i].segment_count = -1;
 	}
+	c->isdbs_ts_id = c->dvbt2_plp_id = -1;
 
 	return 0;
 }
@@ -1488,10 +1489,10 @@
 		c->layer[2].interleaving = tvp->u.data;
 		break;
 	case DTV_ISDBS_TS_ID:
-		c->isdbs_ts_id = tvp->u.data;
+		c->isdbs_ts_id = c->dvbt2_plp_id = tvp->u.data;
 		break;
 	case DTV_DVBT2_PLP_ID:
-		c->dvbt2_plp_id = tvp->u.data;
+		c->dvbt2_plp_id = c->isdbs_ts_id = tvp->u.data;
 		break;
 	default:
 		return -EINVAL;
